libr/core/cmd_anal.c

```diff
@@ -9059,6 +9059,18 @@ static bool is_unknown_file(RCore *core) {
 	return true;
 }
 
+static bool is_apple_target(RCore *core) {
+	const char *arch = r_config_get (core->config, "asm.arch");
+	if (!strstr (arch, "ppc") && !strstr (arch, "arm") && !strstr (arch, "x86")) {
+		return false;
+	}
+	RBinObject *bo = r_bin_cur_object (core->bin);
+	if (bo && bo->plugin && bo->plugin->name) {
```

Comment:
I think we don't need all these checks.
Function `r_bin_object_new` is where `o->plugin` is assigned and there `plugin` is guaranteed to be not-null. Also `plugin->name` should never be null, so these checks should not be necessary (apart from the `if (bo)` check.